### DISCUSSION TITLE
Add a way to restart a task within the new TUI

### DISCUSSION THREAD

**Discussion by ianwalter** [vercel/turborepo#9051] [Ideas]
### Goals

1. A persistent task that has it's own watch / reload may get "stuck" when an error is encountered and it can't reload itself. The only way to restart it is to then ctrl+c and rerun the turbo command that launches the TUI and tasks, but that means killing all persistent tasks and not just the one that got stuck. This is an unpleasant process.

### Background

Zellij has somewhat similar functionality to the Turbo TUI and it lets you restart / rerun a given panel's task with a hotkey (as well as interact with the process much like Turbo currently does).

### Proposal

I think there should be a hotkey like the `r` key that restarts the task much like the `c` key copies the selected text. It would be much smoother than having to rerun all of the tasks.

**Comment by anthonyshew** [comment]
This sound like the default behavior of `turbo watch`: https://turbo.build/repo/docs/reference/watch

Have you given it a try?

**Comment by mcky** [comment]
@anthonyshew 

Watch mode doesn't cover this use case for me

> [Dependency-aware persistent tasks](https://turborepo.com/docs/reference/watch#dependency-aware-persistent-tasks)
> When your script has a built-in watcher (like next dev) capable of detecting changes in dependencies, you don't need to use turbo watch

Next is my use case here - I've got a monorepo with multiple tasks running, but in this case I was updating my env vars and wanted to only restart that next task, not all of them

I know turbo and next both have systems for handling env vars, but other similar situations come up where I've modified something external and need to restart, e.g. my nestjs project reads some external configs on startup

**Reply by tannerbaum** [reply]
Yup that is the exact use case I just ran into
